### DISCUSSION TITLE
Testing suite for the theme

### DISCUSSION THREAD

**Discussion by vedranmiletic** [squidfunk/mkdocs-material#5492] [Ideas]
Since issue #5487 was raised yesterday I had an idea that it would be nice to have a test suite of `mkdocs.yml` files. These files would contain various configuration permutations that would be, in the most basic approach, predefined as "should compile without errors" or "should raise an error such and such". I would be willing to help in producing some representative examples and I am sure others from the community would help with that as well.

Of course, such collection would be just a starting point that would help avoid issues as the one mentioned. It would be better to also check the produced result and verify it matches the expected one. However, that is a much bigger effort.

**Comment by squidfunk** [comment]
Excellent idea – that's been long on my list. The big problem here lies in the endless permutations that can happen, but testing at least a subset would already help, yes. We haven't had the time to further explore it yet, but we really want to build a set of [examples for Material for MkDocs](https://github.com/mkdocs-material/examples). Those examples would be excellent candidates to specify those canonical use cases that should compile without errors, and could serve as a regression suite. Additionally, we could link those examples from our docs in various places, as they would be excellent starting points for bootstrapping new projects with specific features.

I'll talk to @katharinalisalin about this, as she's the one on the forefront of building the examples, and we see how we can coordinate this. Building examples would also be a good candidate for community contributions, as it doesn't involve deep knowledge of the underlying architecture, but provides huge value.
